frog/imports/ui/Repository.jsx

```diff
@@ -1,9 +1,230 @@
+          checked={this.state.plane.indexOf(value) == -1}
+          onClick={this.toggleFilterPlane.bind(this, value)}
+        /> {value}
+      </span>
+    );
+  }
+
+  createFilterBoxType(value) {
+    //Creates the checkboxes for the type filter, the box is checked if the
+    //user wants to show the activities of the given type
+    return (
+      <span>
+        <input
+          type="checkbox"
+          readOnly
+          checked={this.state.type.indexOf(value) == -1}
+          onClick={this.toggleFilterType.bind(this, value)}
+        /> {value}
+      </span>
+    );
+  }
+
+
+
+  handleFilterName(event) {
+    event.preventDefault();
+    const text = ReactDOM.findDOMNode(this.refs.nameFilter).value.trim();
+    this.setState({nameFilterText:text});
+  }
+
+  handleFilterNameSubmit(event) {
+    event.preventDefault();
+  }
+
+  renderListActivities() {
+
+    var typesLowerCase = this.state.type.map(type => type.toLowerCase())
+
+    return (
+      this.props.activities ?
+        this.props.activities
+          //Filters activities with a name containing the text from the nameFilter input.
+          .filter((activity) => activity.name.indexOf(this.state.nameFilterText) != -1)
+          //Filters activities with respect to the list of filters
+          .filter((activity) => this.state.plane.indexOf(activity.plane) == -1)
+          .filter((activity) => typesLowerCase.indexOf(activity.type) == -1)
+          .map ((activity) => (
+            <Activity
+              key={activity._id}
+              id={activity._id}
+              type={activity.type}
+              name={activity.name}
+              plane={activity.plane}
+              object={activity.object} />
+          )) : <li>No activity</li>
+    );
+  }
+
+  render() {
+
+    //To add a new filter for the planes or types, just add the corresponding
+    //checkbox by using a createFilterBox and the rest is taken care of by
+    //the code.
+    return (
+        <div>
+          <h2>Activities:</h2>
+
+          <form className="input-filter-name"
+            onInput={this.handleFilterName.bind(this)}
+            onSubmit={this.handleFilterNameSubmit.bind(this)}>
+
+            <input
+              type="text"
+              ref ="nameFilter"
+              placeholder="Filter by name"
+            /><br/><br/>
+
+            <div className="filters">
+            Show only plane(s):
+              {this.createFilterBoxPlane(1)}
+              {this.createFilterBoxPlane(2)}
+              {this.createFilterBoxPlane(3)}
+              <br/>
+            Show only the following type(s) of activities:
+              {this.createFilterBoxType("Lecture")}
+              {this.createFilterBoxType("Quizz")}
+              {this.createFilterBoxType("Video")}
+            </div>
+
+          </form>
+          <ul>
+            {this.renderListActivities()}
+          </ul>
+        </div>
+      );
+  }
+
+
+}
+
+class Activity extends Component {
```

Comment:
I would suggest having a separate file for this class, simply to factor the code. It's good to start like this now not to end up with 1000 lines long files in three weeks. The best would be to create a new folder `imports/ui/repository` containing `Repository.jsx` and `Activity.jsx`.